src/netlink_namespace.c

```diff
@@ -680,62 +688,68 @@ static int ir_config(const char *key, const char *value) {
   return (status);
 } /* int ir_config */
 
+static void close_fds(int* selfns, int* targetns) {
+  if ((*selfns >= 0) && (close(*selfns) != 0)) {
+    ERROR("Closing \"selfns\" file failed: %m");
+  }
+  if ((*targetns >= 0) && (close(*targetns) != 0)) {
+    ERROR("Closing \"targetns\" file failed: %m");
+  }
+} /* fd_close */
+
 static int ir_init (void)
 {
   int selfns, targetns;
 
   DEBUG("Initializing netlink_namespace plugin\n");
   if (ns_name == NULL) {
     ERROR("netlink_namespace plugin: ir_init: Did not get the namespace name?");
-    return (-1);
-  }
-
-  /* Save "current" namespace status */
-  if (NULL == (nloriginal = mnl_socket_open(NETLINK_ROUTE))) {
-    ERROR("netlink_namespace plugin: ir_init: Unable to open mnl_socket for current namespace  (errno %d)", EXIT_FAILURE);
-    return (-1);
-  }
-
-  if (0 > mnl_socket_bind(nloriginal, 0, MNL_SOCKET_AUTOPID)) {
-    ERROR("netlink_namespace plugin: ir_init: Unable to bind to netlink for current namespace (errno %d)", EXIT_FAILURE);
-    return (-1);
+    return -1;
   }
 
+  /* Save "current" namespace */
   if ((selfns = open(SELFNS, O_RDONLY)) < 0) {
     ERROR("netlink_namespace plugin: ir_init: Could not open self namespace file at '%s'", SELFNS);
-    return (-1);
+    close_fds(&selfns, &targetns);
+    return -1;
   }
 
   /* Save custom configured namespace status */
 
   DEBUG("netlink_namespace plugin: ir_init: Switching to namespace on file '%s'.", ns_name);
   if ((targetns = open(ns_name, O_RDONLY)) < 0) {
     ERROR("Unable to open namespace file '%s'", ns_name);
-    return(-1);
+    close_fds(&selfns, &targetns);
+    return -1;
   }
 
   if (setns(targetns, CLONE_NEWNET) != 0) {
```

Comment:
Also, switching function name to something else, since it's not just closing files anymore.